python/gink/impl/api_app.py

```diff
@@ -0,0 +1,144 @@
+            if not auth_header or auth_header != f"Bearer {self.auth_token}":
+                return self._bad_auth_handler(start_response)
+
+        request_body = environ.get("wsgi.input").read().decode("utf-8")
+        if request_body:
+            try:
+                body = loads(request_body)
+            except JSONDecodeError: # Not valid JSON
+                return self._bad_body_handler(start_response)
+
+        if environ.get("REQUEST_METHOD") == "GET":
+            default = object()
+            result = self.root.get(raw_path.split("/"), default)
+            if result is default:
+                return self._data_not_found_handler(start_response)
+            return self._get_handler(result, start_response)
+
+        elif environ.get("REQUEST_METHOD") == "PUT":
+            try:
+                value = body["value"] # Expecting a body of { "value": "some value" } and an optional comment.
+            except AttributeError:
+                return self._bad_body_handler(start_response)
+
+            content_type = environ.get("HTTP_CONTENT_TYPE")
+            # application/json is handled by default
+            if content_type == "text/plain":
+                value = str(value)
+            elif content_type == "application/octet-stream": # For binary data
+                try:
+                    value = bytes(value, "utf-8")
+                except TypeError:
+                    return self._bad_body_handler(start_response)
+
+            comment = body.get("comment")
+            self.root.set(raw_path.split("/"), value, comment=comment)
+            return self._set_handler(start_response)
+
+        # elif environ.get("REQUEST_METHOD") == "DELETE":
+            # deleted = bool(self.root.delete(raw_path.split("/")))
+            # if deleted:
+            #     return self._delete_handler(start_response)
+            # else:
+            #     return self._data_not_found_handler(start_response)
+
+        else:
+            return self._bad_method_handler(start_response)
+
+    def _get_handler(self, data, start_response):
+        """
+        Default response handler to return gink data in JSON format.
+        """
+        status = '200 OK'
+        content_type = 'application/json'
+        if type(data) != bytes:
+            data = dumps(data).encode()
+        else:
+            content_type = 'application/octet-stream'
+
+        headers = [('Content-type', content_type)]
+        start_response(status, headers)
+        assert type(data) == bytes, "data isn't bytes?"
+        return [data]
+
+    def _set_handler(self, start_response):
+        status = '201 Created'
+        headers = [('Content-type', 'text/plain')]
+        start_response(status, headers)
+        return [b'Entry updated or created.']
+
+    # def _delete_handler(self, start_response):
+    #     status = '200 OK'
+    #     headers = [('Content-type', 'text/plain')]
+    #     start_response(status, headers)
```

Comment:
Delete for directories doesn't support path traversal yet. That can be a separate PR and I will uncomment these lines when that is working.